pulpcore/cli/rpm/repository.py

```diff
@@ -142,8 +143,20 @@ def repository(ctx: click.Context, pulp_ctx: PulpCLIContext, repo_type: str) ->
     click.option(
         "--package-checksum-type", type=click.Choice(CHECKSUM_CHOICES, case_sensitive=False)
     ),
-    click.option("--gpgcheck", type=click.Choice(("0", "1"))),
-    click.option("--repo-gpgcheck", type=click.Choice(("0", "1"))),
+    click.option(
+        "--gpgcheck",
+        type=click.Choice(("0", "1")),
+        help=_(
+            """DEPRECATED:Option specifying whether a client should perform a GPG signature check on packages."""
```

Comment:
What is the deprecation here? Is the option removed from upstream? Can we try to be nice and translate the values into repo_config for the user (while still warning them) given we talk with a new server?

Reply:
The option will be removed, for now just deprecated https://github.com/pulp/pulp_rpm/pull/3298/files#diff-71a2808f74566cf951fcf9b0c3b895baaad58a568b4f26cc6f6ab22bef32d298R158

The values are being translated from gpgcheck into repo_config https://github.com/pulp/pulp_rpm/pull/3298/files#diff-71a2808f74566cf951fcf9b0c3b895baaad58a568b4f26cc6f6ab22bef32d298R161
What exactly you had in  mind?

Reply:
So the translation happens on the api side (for some versions). Then we can just declare the options here to need a version below that. That sounds fine.